### PR TITLE
Big Slappy | Giant Weapon DLC [MERGE READY!!!]

### PR DESCRIPTION
**Pull request by Comxy**
## About The Pull Request
Big Slappy is a new illegal shovel/wrench which can be obtained through illegals means. Parts you need to craft it can be found on the black market uplink for a low price, but if it is not there you can find it on the imports page on your hacked cargo console for 4400 credits usually. Big Slappy makes you slow and you will get 10 recoil damage everytime you hit someone and you will knock down your opponent for two seconds while throwing people back like a baseball bat. It deals 30 damage and is the fastest shovel/wrench out there. It can be folded into its wrench form which will change its typing and makes it look like a wrench. It also deals more damage in the wrench state, but also deals more recoil damage and your target will not get knocked down. This weapon will probably only appear in a few rounds since it has recoil damage, is slow and costs a lot to obtain. It also has a longer meele attack cooldown.

To craft it you need 4 wrenches, 1 drill, 5 plasteel, 10 rods and 1 big slappy parts using tools.
promo
## Why It's Good For The Game
Cool new weapon.
## Changelog
:cl:
add: Added big slappy
image: Added big slappy parts and big slappy sprites
/:cl:

### CODE REVIEW

**Comment by CCC23** [comment]
Using big slappy to obliterate a traitors nuts

File: code/datums/components/item_slowdown.dm [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/// This component is used to make a mob slower when the item the component is attached to is picked up.
+/datum/component/item_slowdown
```
Comment (MrMelbert) [inline code/datums/components/item_slowdown.dm:2]: Item have a `var/slowdown` that slows down the mob when held, why not use that?

Comment (Comxy) [reply code/datums/components/item_slowdown.dm:2]: That doesn't work for some reason, I think I have only seen that variable work on clothes

**Comment by Wallemations** [comment]
How the hell is this a wrench

**Comment by Comxy** [comment]
> How the hell is this a wrench

If you look at the parts, 4 rods are used to secure 4 wrenches to a drill, which is the giant part covered with 2 plasteel. The wrenches under the plasteel spin around like a lawn mower, while the correct bolt size is changed through the big slappy parts to deconstruct whatever needs to be wrenched. The other materials are used for the handle with a small counter weight on the other side to balance the Big Slappy.

File: code/datums/components/item_slowdown.dm (hunk 1 above)
Comment (MrMelbert) [reply code/datums/components/item_slowdown.dm:2]: You need to set item flag `SLOWS_WHILE_IN_HAND` and call `update_equipment_speed_mods`

**Comment by Sealed101** [comment]
THE RETVRN

**Comment by YuliceTheDruid** [comment]
This is not what I meant when I said we should add big slappy

**Comment by LSommers** [comment]
You can't fool me, Tile Remover, I know your inhand
that being said comically large slapstick weapons will always be funny to me
I'd suggest either making it a slapstick weapon or a tool, not both, if only because the sprite looks like it's confused as to what it's meant to be

it looks like a metal snow shovel with an enormous ice pick built onto the end of the handle

actually, make it good at clearing ice and snow from icebox and the sprite would work
"assistant, this is Delta, we don't have any snow, why do you have-"
[bonk]

**Comment by HoodMilk** [comment]
Only a spoonful.

**Comment by Wallemations** [comment]
> > How the hell is this a wrench
> 
> If you look at the parts, 4 rods are used to secure 4 wrenches to a drill, which is the giant part covered with 2 plasteel. The wrenches under the plasteel spin around like a lawn mower, while the correct bolt size is changed through the big slappy parts to deconstruct whatever needs to be wrenched. The other materials are used for the handle with a small counter weight on the other side to balance the Big Slappy.

Yes but how is this, visually, a wrench.
This looks more like a snow shovel than a wrench tbh

**Comment by Farquaar** [comment]
> > > How the hell is this a wrench
> > 
> > 
> > If you look at the parts, 4 rods are used to secure 4 wrenches to a drill, which is the giant part covered with 2 plasteel. The wrenches under the plasteel spin around like a lawn mower, while the correct bolt size is changed through the big slappy parts to deconstruct whatever needs to be wrenched. The other materials are used for the handle with a small counter weight on the other side to balance the Big Slappy.
> 
> Yes but how is this, visually, a wrench. This looks more like a snow shovel than a wrench tbh

Looks like a cyber flyswatter to me

**Comment by Wallemations** [comment]
> Looks like a cyber flyswatter to me
If you told me it was that or a big spatula I'd laugh and say ok sure but a wrench????

File: code/datums/components/item_slowdown.dm (hunk 1 above)
Comment (Comxy) [reply code/datums/components/item_slowdown.dm]: Oh yeah that works, thanks

**Comment by kinneb** [comment]
"only a wrenchful"

**Comment by Exester509** [comment]
Where exactly is the wrench supposed to be?

**Comment by Comxy** [comment]
It has a wrench state now, like the other hybrid tools, it is both a shovel and a wrench.
giant_big_wrench_2

**Comment by Wallemations** [comment]
Ok you've made it from confusing to really funny, good work

**Review by Wallemations — APPROVED** [review]

**Comment by optimumtact** [comment]
it looks bad

**Comment by Comxy** [comment]
Yeah, I cannot believe the people making Big Slappy fucked up that bad, I guess they didn't follow the instructions right

**Comment by CCC23** [comment]
Can I make bigger slappy?

**Comment by Comxy** [comment]
Of course you can make Bigger Slappy

**Comment by Shadowflame909** [comment]
Now I'm truly Dirty Dan

File: code/modules/mining/equipment/mining_tools.dm [hunk 2]
```diff
@@ -283,3 +283,88 @@
...
+
+/obj/item/shovel/giant_wrench/attack(mob/living/target_mob, mob/living/user)
+	..()
+	if(QDELETED(target_mob))
+		return
+	if(do_launch)
+		var/atom/throw_target = get_edge_target_turf(target_mob, get_dir(user, get_step_away(target_mob, user)))
+		target_mob.throw_at(throw_target, 2, 2, user, gentle = TRUE)
+		target_mob.Knockdown(2 SECONDS)
+	user.adjustBruteLoss(force / recoil_factor)
```
Comment (Ghommie) [inline code/modules/mining/equipment/mining_tools.dm:368]:
```suggestion
	var/body_zone = pick(BODY_ZONE_HEAD, BODY_ZONE_CHEST, BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)
	user.apply_damage(force / recoil_factor, BRUTE, body_zone, target.run_armor_check(body_zone, MELEE))
```
I believe armor should be taken into account (you can always lower the blocked damage by multiplying the fourth arg by a 0-1 value).

File: code/game/objects/items/tools/wrench.dm [hunk 3]
```diff
@@ -23,6 +23,8 @@
 	tool_behaviour = TOOL_WRENCH
 	toolspeed = 1
 	armor_type = /datum/armor/item_wrench
+	item_flags = SLOWS_WHILE_IN_HAND
+	slowdown = 3
```
Comment (Ghommie) [inline code/game/objects/items/tools/wrench.dm:27]: Oh no no no, this is the generic wrench type you're modifying. 
```suggestion
```

File: code/modules/mining/equipment/mining_tools.dm [hunk 4]
```diff
@@ -283,3 +283,88 @@
...
+	force = 30
+	throwforce = 20
+	block_chance = 30
+	throw_range = 2
+	demolition_mod = 2
+	armor_type = /datum/armor/giant_wrench
+	resistance_flags = FIRE_PROOF
+	wound_bonus = -10
+	attack_verb_continuous = list("bonks", "bludgeons", "pounds")
+	attack_verb_simple = list("bonks", "bludgeons", "pounds")
```
Comment (Ghommie) [inline code/modules/mining/equipment/mining_tools.dm:309]:
```suggestion
	attack_verb_simple = list("bonk", "bludgeon", "pound")
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
Oh boy, this thing is brutal. 40 damage in wrench mode, plus the knockdown. Nevermind the hefty slowdown and conspicious inhand, if they hit you with this it's game over. I'm not particularly worried about it being the antithesis of the House Edge, but it should have at least a longer click cooldown to reflect how heavy metal it's.

File: code/modules/mining/equipment/mining_tools.dm [hunk 5]
```diff
@@ -283,3 +283,90 @@
...
+/obj/item/shovel/giant_wrench/attack(mob/living/target_mob, mob/living/user)
+	..()
+	if(QDELETED(target_mob))
+		return
+	if(do_launch)
+		var/atom/throw_target = get_edge_target_turf(target_mob, get_dir(user, get_step_away(target_mob, user)))
+		target_mob.throw_at(throw_target, 2, 2, user, gentle = TRUE)
+		target_mob.Knockdown(2 SECONDS)
+	var/body_zone = pick(BODY_ZONE_HEAD, BODY_ZONE_CHEST, BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)
+	user.apply_damage(force / recoil_factor, BRUTE, body_zone, target.run_armor_check(body_zone, MELEE))
```
Comment (Ghommie) [inline code/modules/mining/equipment/mining_tools.dm:370]:
```suggestion
	user.apply_damage(force / recoil_factor, BRUTE, body_zone, user.run_armor_check(body_zone, MELEE))
```
Sowwy.

**Comment by EricZilla** [comment]
It should also knock you down if you miss

**Comment by Ghommie** [comment]
Oh well, I guess I'll merge it as is, so that we can see how actually overpowered this bad boy can be.

**Comment by CCC23** [comment]
holy fuck first shift with this beast and it does 2.5 to kill WHAT THE FUCK

**Comment by optimumtact** [comment]
bad idea to merge this

**Comment by CCC23** [comment]
this was a great idea

**Pull request merged**